src/io.ts

```diff
@@ -414,7 +416,14 @@ export function applyTodoChangesSync(
   }
 
   for (const [fileHash] of remove) {
+    const { dir } = posix.parse(fileHash);
+    const todoDir = posix.join(todoStorageDir, dir);
+
     unlinkSync(posix.join(todoStorageDir, `${fileHash}.json`));
+
+    if (readdirSync(todoDir).length === 0) {
```

Comment:
Do we have to readdirSync here? Don't we know for sure that we do or don't have more?

Reply:
Specifically, the call site knows exactly how many pre-existing items there were and it passes us the removals.

https://github.com/ember-template-lint/ember-template-lint-todo-utils/blob/206fb25c0cbfeb1c6f3ab315bee391ef8ba46bf6/src/io.ts#L204-L212

In this code, if `existing.size === remove.size` we need to delete the directory (and we do not need any additional I/O operations to know that we should delete it).

One possible path forward, would be to update `applyTodoChanges[Sync]` to accept all three maps (add, remove, pre-existing)...

Reply:
That's only true if it's the same file path, not if it isn't. Remember, the subdirs are hashes of the `filePath`, so that's what we're specifically checking for. We don't always necessarily want to remove it.